src/k8s/pkg/k8sd/features/cilium/network.go

```diff
@@ -44,6 +48,13 @@ func ApplyNetwork(ctx context.Context, snap snap.Snap, cfg types.Network, _ type
 		}, nil
 	}
 
+	if err := ensureCniBinDir(ctx, snap.CNIBinDir()); err != nil {
+		return types.FeatureStatus{
+			Version: ciliumAgentImageTag,
+			Message: fmt.Sprintf(networkDeployFailedMsgTmpl, err),
+		}, err
+	}
```

Comment:
As discussed elsewhere, this only runs on one of the nodes in the cluster, and therefore might leave other nodes broken. Perhaps it's best to move this as part of `setup.EnsureAllDirectories()` which runs on all cluster nodes when they join the cluster.